lib/https.c

```diff
@@ -349,6 +350,38 @@ _establish_connection(struct https_request * const req,
 #endif /* HAVE_GETADDRINFO */
 }
 
+/* Provide implementations for HMAC_CTX_new and HMAC_CTX_free when
+ * building for OpenSSL versions older than 1.1.0
+ */
+#if OPENSSL_VERSION_NUMBER < 0x10100000L
+static void EVP_MD_CTX_free(EVP_MD_CTX *ctx)
+{
+    if (ctx != NULL) {
+        EVP_MD_CTX_cleanup(ctx);
```

Comment:
So I think the answer to @xdesai's question and to the crashes you were seeing is: the openssl wiki docs are incorrect (which is ... awesome).

Here's the struct definition for HMAC_CTX in OpenSSL 1.0.2:

```
typedef struct hmac_ctx_st {
    const EVP_MD *md;
    EVP_MD_CTX md_ctx;
    EVP_MD_CTX i_ctx;
    EVP_MD_CTX o_ctx;
    unsigned int key_length;
    unsigned char key[HMAC_MAX_MD_CBLOCK];
} HMAC_CTX;
```

`md_ctx`, `i_ctx`, and `o_ctx` are *not pointers*, so we don't want to attempt to `free()`! I think the right way to do this is probably to change our `EVP_MD_CTX_free()` calls to `EVP_MD_CTX_cleanup()`, and then we don't need to provide an implementation for `EVP_MD_CTX_free()` at all.